<center><img src = 'https://github.com/insaid2018/Term-1/blob/master/Images/INSAID_Full%20Logo.png?raw=true' width="240" height="180"></center>

# EDA On Hotel Booking


## Table of Contents

1. [Introduction](#section1)<br>
2. [Problem Statement](#section2) <br>
3. [Installing & Importing Packages](#section3)<br>
4. [Loading Data](#section4)<br>
  - 3.1 [Description of the Datasets](#section301)<br>
  - 3.2 [Pandas Profiling before Data Preprocessing](#section302)<br>
5. [Data Preprocessing](#section4)<br>
  - 4.1 [Data Preprocessing](#section401)<br>
  - 4.2 [Pandas Profiling after Data Preprocessing](#section402)<br>
6. [Exploratory Data Analysis](#section5)<br>
  - 6.1 [**Analysis based on Outlet metrics**](#section501)<br>
    - 6.1.1 [How many stores are actually owned and run by McDonald's?](#section50101)<br>
    - 6.1.2 [Which outlets generate maximum revenue for the company?](#section50102)<br>
    - 6.1.3 [Top 10 Outlets in terms of Profit](#section50103)<br>
    - 6.1.4 [Outlets with highest Number of Employees](#section50104)<br>
    - 6.1.5 [Where do you find the maximum footfall in McDonald's?](#section50105)<br><br>
  - 6.2 [**Analysis based on Nutritional Value**](#section502)<br>
      - 6.2.1 [Which is the most common category on menu?](#section50201)<br>
      - 6.2.2 [How many calories (energy) on an average are present in each category of McDonald's menu?](#section50202)<br>
      - 6.2.3 [Is there any difference in the nutritional value of a grilled vs crispy chicken?](#section50203)<br>
      - 6.2.4 [On an average, how much sugar consumption happens when you order Desserts, Beverages, Smoothies and Shakes?](#section50204)<br>
      - 6.2.5 [In what sort of foods and beverages do manufacturers include fiber?](#section50205)<br> 
      - 6.2.6 [How to select Nutritious and Non-Nutritious food from menu?](#section50206)<br> 
      - 6.2.7 [Which is the most proteinaceous item on the menu?](#section50207)<br> <br>
  - 6.3 [**Analysis based on Geographical Information**](#section503)<br>
    - 6.3.1 [Which city has the highest number of McDonald's outlets per capita?](#section50301)<br>
    - 6.3.2 [Which state has the highest number of McDonald's outlets per capita?](#section50302)<br>
    - 6.3.3 [Outlet locations in India](#section50303)<br>
    - 6.3.4 [Outlet locations in India](#section50304)<br><br>
  - 5.4 [**Analysis based on Outlet metrics and Nutritional value**](#section504)<br>
    - 5.4.1 [Which outlets have most nutritious item as their best selling item?](#section50401)<br>
    - 5.4.2 [What is the Revenue of the outlet based on the category of its best selling item?](#section50402)<br>
    - 5.4.3 [How much is the Total Selling Price of the outlet based on the category of its best selling item?](#section50403)<br>
    - 5.4.4 [How much is the Total Cost Price of the outlet based on the category of its best selling item?](#section50404)<br><br>
  - 5.5 [**Analysis based on Outlet Metrics and Geographical Information**](#section505)<br>
    - 5.5.1 [Where did McDonald's entered into Joint Venture to start their Outlets?](#section50501)<br>
    - 5.5.2 [Top 10 outlets in India and US based on revenue](#section50502)<br>
    - 5.5.3 [What is the mean revenue of outlets in each US state?](#section50503)<br>
    - 5.5.4 [How much is the Gross Profit Margin of each outlet?](#section50504)<br><br>
  - 5.6 [**Analysis based on Nutritional value and Geographical information**](#section506)<br>
    - 5.6.1 [Comparison of Nutritional content of each category between India and US](#section50601)<br>
    - 5.6.2 [How are Indian menu items compare to US menu items in terms of nutrition?](#section50602)<br><br>
  - 5.7 [**Analysis based on outlet metrics, menu items, and geographical information**](#section507)<br>
    - 5.7.1 [How is the revenue generation based on different menu items in India and US?](#section50701)<br><br>
6. [Conclusion](#section6)
  - 6.1 [Actionable Insights](#section601)

<a id=section1></a>
## 1. Introduction
The dataset contains data from two different hotels. Taj Gateway Resort & The Westin Hotel

Both hotels are located in the state of Haryana, India ("The Westin is located in Gurgaon and The Taj Gateway Resort is located near the Damdama Lake in the souther part of Haryana"). The distance between these two locations is 28 kms by car.

The data contains "bookings due to arrive between the 1st of July of 2015 and the 31st of August 2017".
Note: For most questions I will only use bookings that were not canceled, to get acutal guest numbers. As you will see, this is quite a big difference.



---
<a id=section2></a>
# **2. Problem Statement**
---
How to know if a guest will actually come & book a room which can help the hotel to plan things like personel & food requirements.


---
<a id =section3></a>
# **3. Installing & Importing Libraries**
---

In [6]:
#-------------------------------------------------------------------------------------------------------------------------------
import pandas as pd                                                 # Importing package pandas (For Panel Data Analysis)
from pandas_profiling import ProfileReport                          # Import Pandas Profiling (To generate Univariate Analysis)
#-------------------------------------------------------------------------------------------------------------------------------
import numpy as np                                                  # Importing package numpys (For Numerical Python)
#-------------------------------------------------------------------------------------------------------------------------------
import matplotlib.pyplot as plt                                     # Importing pyplot interface to use matplotlib
import seaborn as sns                                               # Importing seaborn library for interactive visualization
%matplotlib inline
#-------------------------------------------------------------------------------------------------------------------------------
import plotly.express as px                                         # Library for data visualization
import scipy as sp                                                  # Importing library for scientific calculations
import folium                                                       # Library to generate basic statistics about data
#----------------------------------------------------------------------------------------------------------------------

<a name = section4></a>
# 4. Loading Data


In [20]:
# Importing the hotel bookings dataset

df_hotel = pd.read_excel('C://Users//sverma3//Desktop//Term1 Project//hotel_bookings.xlsx')
df_hotel.head()


hotel  is_canceled  lead_time  arrival_date_year  \
0  Taj Gateway Resort            0        342               2015   
1  Taj Gateway Resort            0        737               2015   
2  Taj Gateway Resort            0          7               2015   
3  Taj Gateway Resort            0         13               2015   
4  Taj Gateway Resort            0         14               2015   

  arrival_date_month  arrival_date_week_number  arrival_date_day_of_month  \
0               July                        27                          1   
1               July                        27                          1   
2               July                        27                          1   
3               July                        27                          1   
4               July                        27                          1   

   stays_in_weekend_nights  stays_in_week_nights  adults  ...  deposit_type  \
0                        0                     0       2  ...    No Deposit   
1                        0                     0       2  ...    No Deposit   
2                        0                     1       1  ...    No Deposit   
3                        0                     1       1  ...    No Deposit   
4                        0                     2       2  ...    No Deposit   

   agent company days_in_waiting_list customer_type   adr  \
0    NaN     NaN                    0     Transient   0.0   
1    NaN     NaN                    0     Transient   0.0   
2    NaN     NaN                    0     Transient  75.0   
3  304.0     NaN                    0     Transient  75.0   
4  240.0     NaN                    0     Transient  98.0   

   required_car_parking_spaces  total_of_special_requests  reservation_status  \
0                            0                          0           Check-Out   
1                            0                          0           Check-Out   
2                            0                          0           Check-Out   
3                            0                          0           Check-Out   
4                            0                          1           Check-Out   

  reservation_status_date  
0              2015-07-01  
1              2015-07-01  
2              2015-07-02  
3              2015-07-02  
4              2015-07-03  

[5 rows x 32 columns]

<a id=section301></a>
### Description of the Dataset

- This dataset provides a column-wise description of the dataset



| Column Name                       | Description                                                                           |
| ----------------------------------|:--------------------------------------------------------------------------------------| 
| hotel								| Name of the Hotel																		|
| is_canceled						| Status of booking cancellation														|  | lead_time						   | Time between when a guest makes a reservation and the actual check-in date			   |
| arrival_date_year					| Year of arrival																		|
| arrival_date_month				| Date of Month of arrival																|
| arrival_date_week_number			| Week of arrival																		|
| arrival_date_day_of_month			| Day of Month of arrival																|
| stays_in_weekend_nights			| Frequency of guest staying in weekend nights											|
| stays_in_week_nights				| Frequency of guest staying in weekday nights											|
| adults							| Count of adults																		|
| children							| Count of children																		|
| babies							| Count of Babies																		|
| meal								| Type of meal ordered																	|
| country							| Country of the guest																	|
| market_segment					| Market Segment of the guest															|
| distribution_channel				| Type of distribution channel															|
| is_repeated_guest					| Repeated visit of the guest (Yes / No) 												|
| previous_cancellations			| Count of previous cancellations by guest												|
| previous_bookings_not_canceled	| Count of non-cancelled previous bookings												|
| reserved_room_type				| Type of room reserved by the guest													|
| assigned_room_type				| Type of room assigned to the guest													|
| booking_changes					| Count of changes in the booking														|
| deposit_type						| Type of Deposit made by guest															|
| agent								| Count of bookings made through agent													|
| company							| Count of booking made by company														|
| days_in_waiting_list				| Count of days in waiting list															|
| customer_type						| Type of Customer																		|
| adr								| adr																					|
| required_car_parking_spaces		| Count of car parking spaces booked													|
| total_of_special_requests			| Total count of special requests														|
| reservation_status				| Status of reservation																	|
| reservation_status_date			| Date of reservation																	|

<br>


In [21]:
df_hotel.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 119390 entries, 0 to 119389
Data columns (total 32 columns):
 #   Column                          Non-Null Count   Dtype         
---  ------                          --------------   -----         
 0   hotel                           119390 non-null  object        
 1   is_canceled                     119390 non-null  int64         
 2   lead_time                       119390 non-null  int64         
 3   arrival_date_year               119390 non-null  int64         
 4   arrival_date_month              119390 non-null  object        
 5   arrival_date_week_number        119390 non-null  int64         
 6   arrival_date_day_of_month       119390 non-null  int64         
 7   stays_in_weekend_nights         119390 non-null  int64         
 8   stays_in_week_nights            119390 non-null  int64         
 9   adults                          119390 non-null  int64         
 10  children                        119386 non-null  float64

- ```info``` function gives us the following insights into the df_hotel dataframe:

  - There are a total of **119390 rows** and **32 columns** in the dataframe.
  
  - There are **16 columns** with a **numeric** datatype, **11 columns** with an **object** datatype, **4 columns** with **float** datatype, **1 column** with **datetime** datatype
  
  - There are **0 missing** values in the data.